pkg/i2gw/ingress2gateway.go

```diff
@@ -122,11 +123,13 @@ func ConstructIngressesFromCluster(ctx context.Context, cl client.Client, ingres
 // by their ProviderName.
 //
 // TODO (levikobi): Issue #45 - let users filter by provider name.
-func constructProviders(conf *ProviderConf) map[ProviderName]Provider {
+func constructProviders(conf *ProviderConf, providers []string) map[ProviderName]Provider {
 	providerByName := make(map[ProviderName]Provider, len(ProviderConstructorByName))
 
 	for name, newProviderFunc := range ProviderConstructorByName {
-		providerByName[name] = newProviderFunc(conf)
+		if slices.Contains(providers, string(name)) {
+			providerByName[name] = newProviderFunc(conf)
+		}
 	}
 
 	return providerByName
```

Comment:
Let's say I misspelled one of the provider names and passed ingres-nginx, instead of ingress-nginx. I expect to get an error saying that the provider I passed doesn't exist. So, we'll probably need to return an error as a second return type in this function.

Reply:
I think this should be unit tested as well. A simple table unit-test would be very nice here.